lib/logstash/inputs/kafka.rb

```diff
@@ -20,168 +21,89 @@
+  config :fetch_max_wait_ms, :validate => :string
+  config :fetch_min_bytes, :validate => :string
+  config :group_id, :validate => :string, :default => "ll"
+  config :heartbeat_interval_ms, :validate => :string
+  config :key_deserializer, :validate => :string, :default => "org.apache.kafka.common.serialization.StringDeserializer"
+  config :max_partition_fetch_bytes, :validate => :string
+  config :metadata_max_age_ms, :validate => :string
+  config :metric_reporters, :validate => :string
+  config :metrics_num_samples, :validate => :string
+  config :metrics_sample_window_ms, :validate => :string
+  config :partition_assignment_strategy, :validate => :string
+  config :receive_buffer_bytes, :validate => :string
+  config :reconnect_backoff_ms, :validate => :string
+  config :request_timeout_ms, :validate => :string
+  config :retry_backoff_ms, :validate => :string
+  config :send_buffer_bytes, :validate => :string
+  config :session_timeout_ms, :validate => :string, :default => "30000"
+  config :value_deserializer, :validate => :string, :default => "org.apache.kafka.common.serialization.StringDeserializer"
+  config :num_threads, :validate => :number, :default => 1
+  config :topics, :validate => :array, :required => true
+  config :timeout_ms, :validate => :number, :default => 100
+  config :max_messages, :validate => :number
 
   public
   def register
-    LogStash::Logger.setup_log4j(@logger)
-    options = {
-        :zk_connect => @zk_connect,
-        :group_id => @group_id,
-        :topic_id => @topic_id,
-        :auto_offset_reset => @auto_offset_reset,
-        :rebalance_max_retries => @rebalance_max_retries,
-        :rebalance_backoff_ms => @rebalance_backoff_ms,
-        :consumer_timeout_ms => @consumer_timeout_ms,
-        :consumer_restart_on_error => @consumer_restart_on_error,
-        :consumer_restart_sleep_ms => @consumer_restart_sleep_ms,
-        :consumer_id => @consumer_id,
-        :fetch_message_max_bytes => @fetch_message_max_bytes,
-        :allow_topics => @white_list,
-        :filter_topics => @black_list,
-        :value_decoder_class => @decoder_class,
-        :key_decoder_class => @key_decoder_class
-    }
-    if @reset_beginning
-      options[:reset_beginning] = 'from-beginning'
-    end # if :reset_beginning
-    topic_or_filter = [@topic_id, @white_list, @black_list].compact
-    if topic_or_filter.count == 0
-      raise LogStash::ConfigurationError, 'topic_id, white_list or black_list required.'
-    elsif topic_or_filter.count > 1
-      raise LogStash::ConfigurationError, 'Invalid combination of topic_id, white_list or black_list. Use only one.'
-    end
-    @kafka_client_queue = SizedQueue.new(@queue_size)
-    @consumer_group = create_consumer_group(options)
-    @logger.info('Registering kafka', :group_id => @group_id, :topic_id => @topic_id, :zk_connect => @zk_connect)
+    @message_count = 0
+    @runner_threads = []
   end # def register
 
   public
   def run(logstash_queue)
```

Comment:
Now that we've removed `kafka_client_queue`, just call this `queue` to be consistent with other plugins? wdyt?